File: BeitPatissierServer/Models/Order.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace BeitPatissierServer.Models
{
    public class Order
    {
        public int OrderId { get; set; }
        public DateTime OrderDate { get; set; }
        public int UserId { get; set; }
        [ForeignKey(nameof(UserId))]
        public BPUser User { get; set; }
        public string CustomerName { get; set; }
        public string CustomerEmail { get; set; }
        public string CustomerPhone { get; set; }
        public string ShippingAddress { get; set; }
        public ICollection<OrderLine> OrderLines { get; set; }
        public decimal TotalAmount { get; set; }
    }
}

File: beit-patissier-manage/src/models/Order.ts

import type { OrderLine } from "./OrderLine";

export class Order {
    OrderId: number;
    OrderDate: Date;
    UserId: number;
    CustumerName: string;
    CustumerEmail: string;
    CustumerPhone: string;
    ShippingAddress: string;
    TotalAmount: number;
    OrderLines: OrderLine[];

    constructor(
        OrderId: number,
        OrderDate: Date,
        UserId: number,
        CustumerName: string,
        CustumerEmail: string,
        CustumerPhone: string,
        ShippingAddress: string,
        OrderLines: OrderLine[],
        TotalAmount: number
    ) {
        this.OrderId = OrderId;
        this.OrderDate = OrderDate;
        this.UserId = UserId;
        this.CustumerName = CustumerName;
        this.CustumerEmail = CustumerEmail;
        this.CustumerPhone = CustumerPhone;
        this.ShippingAddress = ShippingAddress;
        this.OrderLines = OrderLines;
        this.TotalAmount = TotalAmount;
    }
}

export const OrderFieldNamesHebrew: Record<keyof Order, string> = {
    OrderId: 'מספר הזמנה',
    OrderDate: 'תאריך',
    UserId: 'מזהה משתמש',
    CustumerName: 'שם לקוח',
    CustumerEmail: 'מייל',
    CustumerPhone: 'טלפון',
    ShippingAddress: 'כתובת למשלוח',
    TotalAmount: 'סך הזמנה',
    OrderLines: 'שורות הזמנה'
};
